### DISCUSSION TITLE
Feature Request: merge branch locally but never commit

### DISCUSSION THREAD

**Discussion by Ankiewicz** [community/community#21678] [New to GitHub]
<p>Working in a large team as FE developer we frequently are doing work while the BE is also working. (2 or more branches actively working). Sometimes we are even working off of an alternate master/feature branch. (which can be 3 or more branches sometimes)</p>
<p>So we tend to have to merge other peoples branches into our current branch in order to develop our work in parallel but then we have to create another branch after that in order to strip out only our code for the PR. </p>
<p>I’ve been looking for the best way to handle this but haven’t found a really prime method. </p>
<p>My current practice is on my branch (all of my work stashed or commited) I  <code>git pull --squash remote/branch</code> the other branch. do my work and commit single files (very rarely the same files from the squashed remote branch). </p>
<p>For updates to remote branch, I commit my work or stash all work in progress, merge new work. <code>git stash apply</code> resolve conflicts and develop.</p>
<p>Pushing my PR, commit only my work (thank you for <code>git add -i</code>) stash remote branch work and PR from there… </p>
<p>Feature request: <br>
The ability to commit/merge/include a branch(remote or local) that actually never gets commited. Would need to allow you to update it, or just create another hidden commit. </p>

**Comment by BelHFL** [comment]
<p>Hi <a class="mention" href="/ankiewicz">@ankiewicz</a> ,</p>
<p>Thanks for this feedback! We’re always working to improve GitHub and the GitHub Community Forum, and we consider every suggestion we receive. I’ve logged your feature request in our internal feature request list. Though I can’t guarantee anything or share a timeline for this, I can tell you that it’s been shared with the appropriate teams for consideration.</p>
<p>Cheers!</p>

**Comment by lee-dohm** [comment, marked as answer]
<p>I’m not entirely sure what it is you’re asking for. You can already create a branch and then never add an additional commit to it. You can also already <a href="https://git-scm.com/docs/git-checkout#Documentation/git-checkout.txt---orphanltnewbranchgt" rel="nofollow noopener">create a branch that has no commits on it using the <code>--orphan</code> option</a>. So I’m not sure what it is you need that doesn’t already exist within git itself.</p>
<p>Additionally, I’m not sure I understand your workflow. It sounds like you have a fairly standard setup:</p>
<ol>
<li>There is a main branch somewhere</li>
<li>There are teams working on their own branches</li>
<li>You are working on some other branch</li>
<li>Occasionally, you need to bring in the latest code from some team branch</li>
<li>When your work is done, you need to merge your code into one of the team branches or the main branch</li>
</ol>
<p>But the process you describe sounds overly complex for the above scenario. Am I understanding your situation correctly?</p>
<p>Final note, I don’t know what the terms FE and BE mean :grinning:</p>
<p>Let me know if you can give more details or have more questions.</p>
